### PR TITLE
lib/web: add package-level logger

### LINKED ISSUE #4110: Define a global logger with prefix [WEB] for web directory
We need to define a global logger for `teleport/libweb/` directory. 

Currently we manually write `[WEB]`:
image

Or some are not defined:
image

Or some might be possibly manually defined per func.

Example:
```
var log = logrus.WithFields(logrus.Fields{
	trace.Component: teleport.ComponentWeb,
})
```

### PR DESCRIPTION
**Pull request by a-palchikov**
Add package-level logger that removes the necessity of augmenting the log messages with an explicit `[web]` prefix.

Fixes https://github.com/gravitational/teleport/issues/4110.
Requires https://github.com/gravitational/teleport.e/pull/168.

### CODE REVIEW

File: go.mod
Comment (a-palchikov) [inline go.mod:94]: This is no longer indirect.

File: lib/web/apiserver.go
Comment (a-palchikov) [inline lib/web/apiserver.go:147]: Moved from `ProxyHostPort` to avoid re-parsing and repetitive error logs if the SSH address is invalid.

**Comment by webvictim** [comment]
Looks like the linter failed in CI: https://drone.gravitational.io/gravitational/teleport/2703/1/4

```
lib/web/apiserver.go:149:3: printf: Warn call has possible formatting directive %v (govet)
--
log.WithError(err).Warn("Invalid SSH proxy address, will use default port %v.", defaults.SSHProxyListenPort)
^
Makefile:285: recipe for target 'lint-go' failed
make: *** [lint-go] Error 1
```

You can run it locally with `make lint` or in Docker with `make -C build.assets lint`.

File: .gitignore
Comment (webvictim) [inline .gitignore:61]: I would move this up to the "editors" section above where `.vscode` is already added to keep everything together.

**Review by webvictim — APPROVED** [review]
LGTM, please wait for review from the other code owners though.

**Review by russjones — COMMENTED** [review]
Not sure about a package level logger.

How attaching `*logrus.Entry` to `Handler` then you can just do `h.log.Debugf(...)` in the log lines. When you need to add additional information, like SAML or OIDC, maybe use `WithFields`?

What do you think?

**Comment by a-palchikov** [comment]
I think either way is fine. Using a package-level implementation might be a convenience or the other way around. I went from the existing code where this was already a case ([here](https://github.com/gravitational/teleport/blob/dmitri/4110-logger/lib/auth/init.go#L48-L50), [here](https://github.com/gravitational/teleport/blob/dmitri/4110-logger/lib/service/service.go#L80-L82) and [here](https://github.com/gravitational/teleport/blob/dmitri/4110-logger/lib/pam/pam.go#L100-L102) are just a couple of examples)

**Comment by a-palchikov** [comment]
@russjones Addressed your concern in https://github.com/gravitational/teleport/pull/4645/commits/c56186801e2600a39cbf18f8b407c5aa6a245f9e

**Review by alex-kovoy — APPROVED** [review]

**Review by russjones — APPROVED** [review]

**Comment by a-palchikov** [comment]
Integration test failure is unrelated to these changes.

**Pull request merged**
